src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/iptables_linux.py

```diff
@@ -0,0 +1,159 @@
+from middlewared.service import Service, accepts, job, CallError
+
+import subprocess
+
+
+V4_FILE = '/data/v4-fw.rules'
+V6_FILE = '/data/v6-fw.rules'
+JOB_LOCK = 'firewall_rules_update'
+
+
+class IptablesService(Service):
+
+    class Config:
+        namespace = 'failover.firewall'
+        private = True
+
+    async def generate_rules(self, data):
+        """
+        Generate a list of iptables and ip6tables rules.
+
+        NOTE:
+            SSH and webUI traffic is always allowed.
+        """
+
+        # rules always start with this line
+        v4rules = ['*filter\n']
+        v6rules = ['*filter\n']
+
+        # the positions of these are important
+        # v4 default section
+        v4rules.insert(1, ':INPUT ACCEPT [0:0]\n')
+        v4rules.insert(2, ':FORWARD ACCEPT [0:0]\n')
+        v4rules.insert(3, ':OUTPUT ACCEPT [0:0]\n')
+
+        # v6 default section
+        v6rules.insert(1, ':INPUT ACCEPT [0:0]\n')
+        v6rules.insert(2, ':FORWARD ACCEPT [0:0]\n')
+        v6rules.insert(3, ':OUTPUT ACCEPT [0:0]\n')
```

Comment:
Would it make sense/clearer to do not have \n to every entry and join them at the end with \n?